vms/platformvm/state/state.go

```diff
@@ -1100,50 +1078,7 @@ func (s *state) ApplyValidatorWeightDiffs(
 			return err
 		}
 	}
-	if err := diffIter.Error(); err != nil {
-		return err
-	}
-
-	// TODO: Remove this once it is assumed that all subnet validators have
-	// adopted the new indexing.
-	for height := prevHeight - 1; height >= endHeight; height-- {
```

Comment:
Note: this does potentially change the values returned for heights that were processed before the new indexing existed. However - those heights are less than the current heights required by the proposervm for all chains (XPC)